.changes/unreleased/Features-20230623-173357.yaml

```
@@ -0,0 +1,6 @@
+kind: Features
+body: Add thread_id context var
+time: 2023-06-23T17:33:57.412102-07:00
+custom:
+  Author: NiallRees
+  Issue: "7942"
```

The PR is #7942 and the issue is #7941
```suggestion
  Issue: "7941"
```